pkg/configurator/client.go

```diff
@@ -250,6 +257,20 @@ func meshConfigUpdatedMessageHandler(psubMsg *events.PubSubMessage) {
 	triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingEndpoint != newMeshConfig.TracingEndpoint)
 	triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingPort != newMeshConfig.TracingPort)
 
+	triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.inboundExternalAuth.Enable != newMeshConfig.inboundExternalAuth.Enable)
+	if newMeshConfig.inboundExternalAuth.Enable && !triggerGlobalBroadcast {
```

Comment:
Why is this conditional needed? Why can't we update if the previous config does not match the new one as evaluated in L260?

At the very least, some comments are required here since the code looks a bit convoluted.

Reply:
I'll add a comment. It's just a small optimization, if extAuthz remains disabled, there's no point in checking or triggering updates because the inner config changed. Same thing should be done with the tracing section.

Reply:
That optimization makes sense. Carving these into checks might facilitate easier unit testing.